packages/astro/src/content/types-generator.ts

```diff
@@ -419,6 +423,29 @@ async function writeContentFiles({
 				for (const entryKey of Object.keys(collection.entries).sort()) {
 					const dataType = collectionConfig?.schema ? `InferEntrySchema<${collectionKey}>` : 'any';
 					dataTypesStr += `${entryKey}: {\n	id: ${entryKey};\n  collection: ${collectionKey};\n  data: ${dataType}\n};\n`;
+					if(settings.config.experimental.contentCollectionJSONSchema) {
+						if (collectionConfig) {
+							// writing inside loop makes sense for the json schema, since we want to have a file per collection
+							const collectionSchemasDir = new URL('./schemas/collections/', contentPaths.cacheDir);
```

Comment:
How do we feel about `.astro/schemas/collections/NAME.json`?

Reply:
Looks good. I would remove `collections`, for brevity, unless we predict to have more schemas. If so, they I would invert the names: `.astro/collections/<NAME>.schema.json`

Reply:
That's a good question. I'm not sure if we predict to have more schemas, but I guess we might, so maybe keeping it safer now? 🤔

Reply:
I don't have particular preferences, even your initial suggestion is fine

Reply:
I like your suggestion of having `collections` first, but I'm unsure about `.schema.json`. Any other opinions? 
cc @natemoo-re

Reply:
I think `collections/*` and `*.schema.json` are great!

Reply:
Sounds good to me!